enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/interceptor/CustomStateMachineInterceptor.java

```diff
@@ -103,21 +114,38 @@ public StateContext<EnrollmentState, EnrollmentEvent> postTransition(StateContex
             return context;
         }
 
-        EnrollmentState expectedState;
-        try {
-            expectedState = enrollmentStateProvider.findByPhaseAndStatus(identityVerification.getPhase(), identityVerification.getStatus());
-        } catch (IdentityVerificationException e) {
-            logger.error("Failed post transition check: {}, {}", e.getMessage(), identityVerification);
-            return context;
-        }
-
-        if (expectedState != targetState) {
-            logger.error("Unexpected targetState={} when expectedState={}, {}", targetState, expectedState, identityVerification);
-        } else {
-            logger.debug("Transition to targetState={}, {}", targetState, identityVerification);
-        }
+        logger.debug("Transition to targetState={}, {}", targetState, identityVerification);
 
         return context;
     }
 
+    @Override
+    public void postStateChange(
+            final State<EnrollmentState, EnrollmentEvent> state,
+            final Message<EnrollmentEvent> message,
+            final Transition<EnrollmentState, EnrollmentEvent> transition,
+            final StateMachine<EnrollmentState, EnrollmentEvent> stateMachine,
+            final StateMachine<EnrollmentState, EnrollmentEvent> rootStateMachine) {
+
+        final OwnerId ownerId = message.getHeaders().get(EventHeaderName.OWNER_ID, OwnerId.class);
+        Assert.notNull(ownerId, "ownerId must not be null");
+        final IdentityVerificationEntity identityVerification = identityVerificationService.findByOptional(ownerId).orElseThrow(() ->
```

Comment:
Need to reload, in the message there is an old value and for example update of `sessionInfo` was lost.